.NET Core 3.0 Preview behaviour for --output differs

--output is relative to the current working directory instead of the .csproj file. This does not seem to be documented here or under What's New

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 211afdc7-c796-202d-239d-823f2a499dff
* Version Independent ID: 9439897e-dcb3-bc2e-5cae-35c1ca2e0681
* Content: [dotnet publish command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore21)
* Content Source: [docs/core/tools/dotnet-publish.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-publish.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

Due to this change, I'm unable to build my docker image:
**Dockerfile**:

```
FROM mcr.microsoft.com/dotnet/core/sdk:3.0-alpine as build
WORKDIR /app

COPY helloworldapi/*.csproj .
RUN dotnet restore -r linux-musl-x64

COPY . .
RUN dotnet publish -c Release -r linux-musl-x64 -o out --no-restore

FROM mcr.microsoft.com/dotnet/core/aspnet:3.0-alpine as runtime
WORKDIR /app
EXPOSE 80
COPY --from=build /app/out ./
ENTRYPOINT ["dotnet","./helloworldapi.dll"]
```


**Error**: 
```
/usr/share/dotnet/sdk/3.0.100/Microsoft.Common.CurrentVersion.targets(4563,5): error MSB3024: Could not copy the file "/app/obj/Release/netcoreapp3.0/linux-musl-x64/helloworldapi" to the destination file "bin/Release/netcoreapp3.0/linux-musl-x64/helloworldapi", because the destination is a folder instead of a file. To copy the source file into a folder, consider using the DestinationFolder parameter instead of DestinationFiles. [/app/helloworldapi.csproj]
The command '/bin/sh -c dotnet publish -c Release -r linux-musl-x64 -o test --no-restore' returned a non-zero code: 1
```

This was working in 2.2 but in 3.0 it fails.

Is there any workaround?

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -88,9 +88,21 @@ The `dotnet publish` command's output is ready for deployment to a hosting syste
 
 - **`-o|--output <OUTPUT_DIRECTORY>`**
 
-  Specifies the path for the output directory. If not specified, it defaults to *./bin/[configuration]/[framework]/publish/* for a runtime-dependent executable and cross-platform binaries. It defaults to *./bin/[configuration]/[framework]/[runtime]/publish/* for a self-contained executable.
+  Specifies the path for the output directory.
+  
+  If not specified, it defaults to *[project_file_folder]./bin/[configuration]/[framework]/publish/* for a runtime-dependent executable and cross-platform binaries. It defaults to *[project_file_folder]/bin/[configuration]/[framework]/[runtime]/publish/* for a self-contained executable.
 
-  If the path is relative, the output directory generated is relative to the project file location, not to the current working directory.
+  - .NET Core 3.x SDK and later
+  
+    If a relative path is specified when publishing a project, the output directory generated is relative to the current working directory, not to the project file location.
+
+    If a relative path is specified when publishing a solution, all output for all projects go into the specified folder relative to the current working directory. To make publish output go to separate folders for each project, specify a relative path by using the msbuild `PublishDir` property instead of the `--output` option. For example, `dotnet publish -p:PublishDir=.\publish` sends publish output for each project to a `publish` folder under the folder that contains the project file.
+
+  - .NET Core 2.x SDK
+  
+    If a relative path is specified when publishing a project, the output directory generated is relative to the project file location, not to the current working directory.
+
+    If a relative path is specified when publishing a solution, each project's output goes into a separate folder relative to the project file location. If an absolute path is specified when publishing a solution, all publish output for all projects goes into the specified folder.
 
 - **`--self-contained [true|false]`**
 

